ablySpec/RestChannel.swift

```diff
@@ -198,14 +212,203 @@ class RestChannel: QuickSpec {
+                        if message.payload.encoding.rangeOfString("cipher") == nil {
                             expect(message.content() as? NSObject).to(equal(fixtureMessage["data"].object as? NSObject))
                         }
                     }
                 }
             }
         }
+
+        describe("message encoding") {
+
+            struct TestCase {
+                let value: AnyObject?
+                let expected: JSON
+            }
+
+            let text = "John"
+            let integer = "5"
+            let decimal = "65.33"
+            let dictionary = ["number":3, "name":"John"]
+            let array = ["John", "Mary"]
+            let data = NSString(string: "123456").dataUsingEncoding(NSUTF8StringEncoding)!
+
+            // RSL4a
+            it("payloads should be binary, strings, or objects capable of JSON representation") {
+                let validCases = [
+                    TestCase(value: nil, expected: JSON([:])),
+                    TestCase(value: text, expected: JSON(["data": text])),
+                    TestCase(value: integer, expected: JSON(["data": integer])),
+                    TestCase(value: decimal, expected: JSON(["data": decimal])),
+                    TestCase(value: dictionary, expected: JSON(["data": (dictionary as NSDictionary).toBase64, "encoding": "json/base64"])),
+                    TestCase(value: array, expected: JSON(["data": (array as NSArray).toBase64, "encoding": "json/base64"])),
+                    TestCase(value: data, expected: JSON(["data": data.toBase64, "encoding": "base64"])),
+                ]
+
+                client.httpExecutor = mockExecutor
+
+                validCases.forEach { caseTest in
+                    waitUntil(timeout: testTimeout) { done in
+                        channel.publish(caseTest.value, callback: { error in
+                            expect(error).to(beNil())
+
+                            if let request = mockExecutor.requests.last,
+                                let http = request.HTTPBody {
+                                    expect(caseTest.expected.rawValue as? NSDictionary).to(equal(JSON(data: http).rawValue as? NSDictionary))
+                            }
+                            done()
+                        })
+                    }
+                }
+
+                let invalidCases = [5, 56.33, NSDate()]
+
+                invalidCases.forEach { caseItem in
+                    waitUntil(timeout: testTimeout) { done in
+                        expect { channel.publish(caseItem, callback: nil) }.to(raiseException())
+                        done()
+                    }
+                }
+            }
+
+            // RSL4b
+            it("encoding attribute should represent the encoding(s) applied in right to left") {
+                let encodingCases = [
+                    TestCase(value: text, expected: nil), //FIXME: Should be UTF-8 ?!
```

Comment:
I believe this is in fact correct, see http://docs.ably.io/client-lib-development-guide/features/#RSL4d2 and https://github.com/ably/ably-js/pull/118#issuecomment-137590035